ui/src/asm_cleanup.rs

```diff
@@ -73,7 +74,8 @@ pub fn filter_asm(block: &str) -> String {
     let mut current_label: &str = "";
     let mut line_info: Vec<LineType> = Vec::new();
     let mut labels: HashSet<&str> = HashSet::new();
-    let mut opcode_operands: HashSet<&str> = HashSet::new();
+    let mut opcode_operands: HashSet<&str> = HashSet::new(); 
+    let mut data_dir_label_refs: HashMap<&str, HashSet<&str>> = HashMap::new(); 
```

Comment:
FWIW, you don't have to change this in this PR, but I'll probably go though here with Clippy and remove a bunch of the explicit types on variable declarations — they aren't very idiomatic.

Reply:
Removed.